### ISSUE TITLE
Bug: compute_duration() calls is_date(), which is not in NAMESPACE

### ISSUE THREAD

**Issue by fshanlee** [pharmaverse/admiral#1582]
### What happened?

A testthat script failed, I think because compute_duration() calls is_date(), which is not in the NAMESPACE:

```
Error (test-derive_vars_aage.R:8:3): duration and unit variable are added
Error: Problem while computing `AAGE = compute_duration(...)`.
Caused by error in `is_date()`:
! could not find function "is_date"
Backtrace:
  1. admiraldev::expect_dfs_equal(...)
       at test-derive_vars_aage.R:8:2
 11. admiral::compute_duration(...)
```
I found an is_date() function in a package called "GGally", but it's not exported from that package:

```
> GGally::is_date("foo")
Error: 'is_date' is not an exported object from 'namespace:GGally'
> GGally:::is_date("foo")
[1] FALSE
```
I did not previously see this error whilst running the same testthat script, so I'm wondering what's changed. Was there previously an `is_date()` function that was defined within {admiraldev}?
@bms63 , FYI, this issue needs to be resolved before we can merge pull request https://github.com/pharmaverse/admiral/pull/1550.

**Comment by bms63** [comment]
Hey @sadchla-codes do you have time to look into this?

**Comment by sadchla-codes** [comment]
Yes, I'll look into it

**Comment by fshanlee** [comment]
Hi @bms63 and @sadchla-codes , I just took a look at the **Main** branch of {admiraldev} and answered my own question: `is_date()` was defined in _is.R_.

**Comment by bms63** [comment]
We also just did someone updates around the `is.R` file - so take a look at `devel` as well.

**Comment by fshanlee** [comment]
> We also just did someone updates around the `is.R` file - so take a look at `devel` as well.

`is_date()` isn't defined in _is.R_ in the **devel** branch.

**Comment by sadchla-codes** [comment]
@fshanlee `is_date()` has been replaced with `assert_date_vector()` in devel

**Comment by fshanlee** [comment]
> @fshanlee `is_date()` has been replaced with `assert_date_vector()` in devel

@bms63 , may I create a new pull request to amend `compute_duration() `to call `assert_date_vector()` instead of `is_date()`, and link the new pull request to this issue?

**Comment by bms63** [comment]
that would be awesome.  Can you also scan the code to see if `is_Date()` is used anywhere else?  I thought we got them all!!

**Comment by fshanlee** [comment]
My bad for not pulling the latest {admiral} _devel_ branch from GitHub before running _testthat_: the calls to `is_date()` have already been replaced by calls to `assert_date_vector()` in compute_duration.R. However, I did a scan of all code as suggested, and `is_date()` is still called once in derive_date_vars.R, so I will still create a pull request to change it to `assert_date_vector()`.

 ```
 if (lubridate::is.POSIXct(dt)) {
    return(dt)
  } else {
    if (is_date(dt)) {
      dt <- format(dt, "%Y-%m-%d")
    }
```

**Comment by bundfussr** [comment]
@fshanlee , at this place `is_date()` should be replaced with `is.instant()` because it should not issue an error if it is not a date. I have already updated it in https://github.com/pharmaverse/admiral/pull/1583.

**Comment by fshanlee** [comment]
I just scanned `derive_date_vars.R` for `is_date()` and it is no longer there, so I assume it has been amended since this issue was raised. Therefore, this issue is now resolved, and I am closing it.
